### PR TITLE
remove egd

### PR DESCRIPTION
**Pull request by reaperhulk**
EGD is bad and old and we removed it from pyOpenSSL in 16.0.

### CODE REVIEW

File: src/cryptography/hazmat/bindings/openssl/_conditional.py [hunk 1]
```diff
@@ -75,14 +75,6 @@ def cryptography_has_rsa_r_pkcs_decoding_error():
     ]
 
 
-def cryptography_has_egd():
```
Comment (alex) [inline src/cryptography/hazmat/bindings/openssl/_conditional.py]: Remove from the dict below as well

File: src/_cffi_src/openssl/rand.py [hunk 2]
```diff
@@ -9,7 +9,6 @@
 """
 
 TYPES = """
-static const long Cryptography_HAS_EGD;
```
Comment (alex) [inline src/_cffi_src/openssl/rand.py]: Should we leave this constant but set to `0`? Then there's no plausible backwards compat concern :-)

Comment (reaperhulk) [inline src/_cffi_src/openssl/rand.py]: Sure, why not. At some point it'd be good to remove it entirely but the two lines aren't going to be a maintenance burden in any way...

**Comment by alex** [comment]
Landing despite the coverage because it appears to be a bug.

**Pull request merged**